ext/store/bigquery/client.go

```diff
@@ -35,29 +39,127 @@ func NewClient(ctx context.Context, svcAccount string) (*BqClient, error) {
 		return nil, errors.InternalError(store, "failed to create BQ client", err)
 	}
 
-	return &BqClient{bq: c}, nil
+	return &BqClient{c}, nil
 }
 
 func (c *BqClient) DatasetHandleFrom(ds Dataset) ResourceHandle {
-	dsHandle := c.bq.DatasetInProject(ds.Project, ds.DatasetName)
+	dsHandle := c.DatasetInProject(ds.Project, ds.DatasetName)
 	return NewDatasetHandle(dsHandle)
 }
 
 func (c *BqClient) TableHandleFrom(ds Dataset, name string) TableResourceHandle {
-	t := c.bq.DatasetInProject(ds.Project, ds.DatasetName).Table(name)
+	t := c.DatasetInProject(ds.Project, ds.DatasetName).Table(name)
 	return NewTableHandle(t)
 }
 
 func (c *BqClient) ExternalTableHandleFrom(ds Dataset, name string) ResourceHandle {
-	t := c.bq.DatasetInProject(ds.Project, ds.DatasetName).Table(name)
+	t := c.DatasetInProject(ds.Project, ds.DatasetName).Table(name)
 	return NewExternalTableHandle(t)
 }
 
+func (c *BqClient) BulkGetDDLView(ctx context.Context, dataset Dataset, names []string) (map[job.ResourceURN]string, error) {
+	queryContent := buildGetDDLQuery(dataset.Project, dataset.DatasetName, names...)
+	queryStatement := c.Client.Query(queryContent)
+	rowIterator, err := queryStatement.Read(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	urnToDDL := map[job.ResourceURN]string{}
+	var errorMessages []string
+	for {
+		var values []bigquery.Value
+		if err := rowIterator.Next(&values); err != nil {
+			if errors.Is(err, iterator.Done) {
+				break
+			}
+
+			errorMessages = append(errorMessages, err.Error())
+			continue
+		}
+
+		if len(values) == 0 {
+			continue
+		}
+
+		name, ddl, err := getViewDDL(values)
+		if err != nil {
+			errorMessages = append(errorMessages, err.Error())
+			continue
+		}
+
+		resourceURN := job.ResourceURN("bigquery://" + fmt.Sprintf("%s:%s.%s", dataset.Project, dataset.DatasetName, name))
+		urnToDDL[resourceURN] = ddl
+	}
+
+	if len(errorMessages) > 0 {
```

Comment:
Please have a look if we can use multi-error, as it already has everything we are creating here.